hepcrawl/spiders/desy_spider.py

```diff
@@ -100,27 +90,22 @@ def __init__(
         self.s3_input_bucket = s3_input_bucket
         self.s3_output_bucket = s3_output_bucket
 
-        self.s3_enabled = False if self.source_folder else True
-
-        if self.s3_enabled:
-            if not (
-                    self.s3_server
-                    and self.s3_key
-                    and self.s3_secret
-                    and self.s3_input_bucket
-                    and self.s3_output_bucket
-            ):
-                raise Exception("Missing s3 connection parameters")
-            else:
-                self.s3_connections = self.connect_to_s3()
-                self.s3_resource = self.get_s3_resource()
+        if not (
+                self.s3_server
+                and self.s3_key
+                and self.s3_secret
+                and self.s3_input_bucket
+                and self.s3_output_bucket
+        ):
+            raise Exception("Missing s3 connection parameters")
+        else:
```

Comment:
I think you can remove the `else` here due to the `raise` right above.

Reply:
:+1: